dpnp/dpnp_iface_arraycreation.py

```diff
@@ -1350,12 +1356,14 @@ def tril(x1, k=0):
 
     """
 
-    x1_desc = dpnp.get_dpnp_descriptor(x1, copy_when_nondefault_queue=False)
-    if x1_desc:
-        if not isinstance(k, int):
-            pass
-        else:
-            return dpnp_tril(x1_desc, k).get_pyobj()
+    if not dpnp.isarray(x1):
+        pass
+    elif x1.ndim < 2:
+        pass
+    elif not isinstance(k, int):
```

Comment:
I think it'd be better to implement the same check through `operator.index` like dpctl did.
Then the following code will be supported:
```
a = dpnp.arange(9, dtype=dpnp.int64)
dpnp.tril(a.reshape((3, 3)), k=a[0])
```

Reply:
I try and see:
```python
a = dpnp.arange(9, dtype=dpnp.int64)
operator.index(a[0])
TypeError: 'dpnp_array' object cannot be interpreted as an integer
```

Reply:
It will be supported once PR #1285 is merged.